src/test/lib/ddl/ddl_statement_test.cpp

```diff
@@ -0,0 +1,143 @@
+  auto sql_pipeline = SQLPipelineBuilder{_create_index_single_column}.create_pipeline();
+
+  const auto& [pipeline_status, table] = sql_pipeline.get_result_table();
+  EXPECT_EQ(pipeline_status, SQLPipelineStatus::Success);
+
+  auto targeted_table = Hyrise::get().storage_manager.get_table("table_a");
+  auto column_ids = std::make_shared<std::vector<ColumnID>>();
+  column_ids->emplace_back(ColumnID{0});
+
+  auto actual_index = targeted_table->indexes_statistics().at(0);
+
+  EXPECT_TRUE(actual_index.name == "myindex");
+  EXPECT_TRUE(actual_index.column_ids == *column_ids);
+
+  auto chunk_count = targeted_table->chunk_count();
+  for(ChunkID id=ChunkID{0}; id < chunk_count; id+=1) {
+    auto current_chunk = targeted_table->get_chunk(id);
+    auto actual_indices = current_chunk->get_indexes(*column_ids);
+    EXPECT_TRUE(actual_indices.size() == 1);
+  }
+}
+
+TEST_F(DDLStatementTest, CreateIndexMultiColumn) {
+  auto sql_pipeline = SQLPipelineBuilder{_create_index_multi_column}.create_pipeline();
+
+  const auto& [pipeline_status, table] = sql_pipeline.get_result_table();
+  EXPECT_EQ(pipeline_status, SQLPipelineStatus::Success);
+
+  auto targeted_table = Hyrise::get().storage_manager.get_table("table_a");
+  auto column_ids = std::make_shared<std::vector<ColumnID>>();
+  column_ids->emplace_back(ColumnID{0});
+  column_ids->emplace_back(ColumnID{1});
+
+  auto actual_index = targeted_table->indexes_statistics().at(0);
+
+  EXPECT_TRUE(actual_index.name == "myindex");
+  EXPECT_TRUE(actual_index.column_ids == *column_ids);
+
+  auto chunk_count = targeted_table->chunk_count();
+  for(ChunkID id=ChunkID{0}; id < chunk_count; id+=1) {
+    auto current_chunk = targeted_table->get_chunk(id);
+    auto applied_indices = current_chunk->get_indexes(*column_ids);
+    EXPECT_TRUE(applied_indices.size() == 1);
+  }
+}
+
+TEST_F(DDLStatementTest, CreateIndexWithoutName) {
+  auto sql_pipeline = SQLPipelineBuilder{"CREATE INDEX ON table_a (a)"}.create_pipeline();
+
+  const auto& [pipeline_status, table] = sql_pipeline.get_result_table();
+  EXPECT_EQ(pipeline_status, SQLPipelineStatus::Success);
+
+  auto targeted_table = Hyrise::get().storage_manager.get_table("table_a");
+  auto column_ids = std::make_shared<std::vector<ColumnID>>();
+  column_ids->emplace_back(ColumnID{0});
+
+  auto actual_index = targeted_table->indexes_statistics().at(0);
+
+  EXPECT_TRUE(actual_index.name == "table_a_a");
+  EXPECT_TRUE(actual_index.column_ids == *column_ids);
+
+  auto chunk_count = targeted_table->chunk_count();
+  for(ChunkID id=ChunkID{0}; id < chunk_count; id+=1) {
+    auto current_chunk = targeted_table->get_chunk(id);
+    auto applied_indices = current_chunk->get_indexes(*column_ids);
+    EXPECT_TRUE(applied_indices.size() == 1);
+  }
+}
+
+TEST_F(DDLStatementTest, CreateIndexIfNotExistsRun) {
+  auto sql_pipeline = SQLPipelineBuilder{"CREATE INDEX myindex IF NOT EXISTS ON table_a (a)"}.create_pipeline();
```

Comment:
This is actually incorrect syntax. The correct syntax would be the following:
`CREATE INDEX IF NOT EXISTS myindex ON table_a (a, b)`

https://www.postgresql.org/docs/current/sql-createindex.html